fpsd/features.py

```diff
@@ -0,0 +1,768 @@
+                 "FROM public.current_bursts')                   "
+                 "AS ct(exampleid bigint,                        "
+                 "{}));").format(', '.join(column_names))
+
+        self.engine.execute(query)
+        return "features.burst_lengths"
+
+    def generate_burst_tables(self):
+        self.create_bursts()
+        self.generate_table_burst_length_aggregates()
+        self.generate_table_binned_bursts()
+        self.generate_table_burst_lengths()
+
+        return ["features.burst_length_aggregates",
+                "features.burst_binned_lengths",
+                "features.burst_lengths"]
+
+    def _get_column_tables_of_table(self, schema_name, table_name):
+        query = ("SELECT column_name FROM information_schema.columns "
+                 "WHERE table_schema='{}' "
+                 "AND table_name='{}'").format(schema_name,
+                                               table_name)
+        result = self.engine.execute(query)
+        colnames = []
+        for row in result:
+            colnames.append(row[0])
+        return colnames
+
+    def create_master_feature_view(self, feature_table_names):
+        """This generates a view of all the feature tables at
+        features.frontpage_features in the following form:
+
+        exampleid  | feature_1 | feature_2 ...    | feature_n
+        (integer)  | (bigint)  | (bigint)  ...    | (bigint)
+        --------------------------------------------------------
+        9          | 1         | 2         ...    | 34
+        10         | 2         | 2         ...    | 21
+        11         | 1         | 3         ...    | 36
+
+        A single row represents a single example along with the features
+        for that example. The purpose of the view is to provide a simple
+        way to select train and test sets.
+
+        Args:
+            feature_table_names [list of strings]: list of tables that
+            contain features that we would like to put into the view
+        """
+
+        master_features = {}
+
+        for schema_and_table in feature_table_names:
+            schema_name = schema_and_table.split('.')[0]
+            table_name = schema_and_table.split('.')[1]
+            table_columns = self._get_column_tables_of_table(schema_name,
+                                                             table_name)
+            table_columns.remove("exampleid")
+            master_features.update({schema_and_table: table_columns})
+
+        columns_to_select = "foo.exampleid"
+        full_join_query = ("FROM ( (SELECT exampleid FROM      " 
+                           "features.undefended_frontpage_examples) foo ")
+
+        for table_num, table_name in enumerate(list(master_features.keys())):
+            prefix = 't{}.'.format(table_num)
+            prefixed_columns = [prefix + s for s in master_features[table_name]]
+            columns_to_select = columns_to_select + ', ' + ', '.join(prefixed_columns)
```

Comment:
I think in this case because we can't read the query altogether anyway, it's better if you just don't create the `columns_to_select` var and instead just use `SELECT *` in the uppermost parent query of the `create_new_view` var. See my comments there as well.